src/main/java/net/earthcomputer/clientcommands/command/WaypointCommand.java

```diff
@@ -0,0 +1,166 @@
+
+    private static final DynamicCommandExceptionType ALREADY_EXISTS_EXCEPTION = new DynamicCommandExceptionType(name -> Text.translatable("commands.cwaypoint.alreadyExists", name));
+    private static final DynamicCommandExceptionType NOT_FOUND_EXCEPTION = new DynamicCommandExceptionType(name -> Text.translatable("commands.cwaypoint.notFound", name));
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(literal("cwaypoint")
+            .then(literal("add")
+                .then(argument("name", word())
+                    .then(argument("pos", blockPos())
+                        .executes(ctx -> add(ctx.getSource(), getString(ctx, "name"), getCBlockPos(ctx, "pos")))
+                        .then(argument("dimension", dimension())
+                            .executes(ctx -> add(ctx.getSource(), getString(ctx, "name"), getCBlockPos(ctx, "pos"), getCDimensionArgument(ctx, "dimension").getRegistryKey()))))))
+            .then(literal("remove")
+                .then(argument("name", word())
+                    .suggests((ctx, builder) -> {
+                        Map<String, Pair<BlockPos, RegistryKey<World>>> worldWaypoints = waypoints.get(getWorldIdentifier(ctx.getSource()));
+                        return CommandSource.suggestMatching(worldWaypoints != null ? worldWaypoints.keySet() : Collections.emptySet(), builder);
+                    })
+                    .executes(ctx -> remove(ctx.getSource(), getString(ctx, "name")))))
+            .then(literal("edit")
+                .then(argument("name", word())
+                    .suggests((ctx, builder) -> {
+                        Map<String, Pair<BlockPos, RegistryKey<World>>> worldWaypoints = waypoints.get(getWorldIdentifier(ctx.getSource()));
+                        return CommandSource.suggestMatching(worldWaypoints != null ? worldWaypoints.keySet() : Collections.emptySet(), builder);
+                    })
+                    .then(argument("pos", blockPos())
+                        .executes(ctx -> edit(ctx.getSource(), getString(ctx, "name"), getCBlockPos(ctx, "pos")))
+                        .then(argument("dimension", dimension())
+                            .executes(ctx -> edit(ctx.getSource(), getString(ctx, "name"), getCBlockPos(ctx, "pos"), getCDimensionArgument(ctx, "dimension").getRegistryKey()))))))
+            .then(literal("list")
+                .executes(ctx -> list(ctx.getSource()))
+                .then(argument("current", bool())
+                    .executes(ctx -> list(ctx.getSource(), getBool(ctx, "current"))))));
+    }
+
+    private static String getWorldIdentifier(FabricClientCommandSource source) {
+        String worldIdentifier;
+        if (source.getClient().isIntegratedServerRunning()) {
+            worldIdentifier = source.getClient().getServer().getSaveProperties().getLevelName();
```

Comment:
Would this make waypoints disappear from a world if its name is changed?

Reply:
That's a good thing to keep in mind for when I actually implement saving to a file.